### PR TITLE
Autoscale: Add preview release 3.9.0

### LINKED ISSUE #1330: Add the ability to create, query, modify autopilot containers
**Is your feature request related to a problem? Please describe.**
Autopilot is a preview feature but once it goes GM we will need a way to manage Autopilot containers and databases using the SDK, and there is not open feature to implement that.

**Describe the solution you'd like**
Add SDK support to;
- Create an autopilot container
- Change an autopilot container RU thresholds
- Switch between autopilot and manual provisioning

**Describe alternatives you've considered**
Manually creating containers through the portal is an option but prone to user error.

**Additional context**
Not really required until GM, but being able to do testing before that happens would be useful.

### PR DESCRIPTION
**Pull request by j82w**
# Pull Request Template
## Description

Bumping the preview version to 3.9.0-preview

Updating autoscale to api and bumping the preview release.

3 9 0-preview

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

closes #1330, #1365

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs [hunk 1]
```diff
@@ -173,14 +173,24 @@ public abstract class Database
...
         /// <code language="c#">
         /// <![CDATA[
-        /// ThroughputProperties throughput = await this.cosmosDatabase.ReplaceThroughputPropertiesAsync(10000);
+        /// ThroughputResponse throughput = await this.cosmosDatabase.ReplaceThroughputAsync(
+        ///     ThroughputProperties.CreateFixedThroughput(10000));
+        /// ]]>
+        /// </code>
+        /// </example>
+        /// <example>
+        /// The following example shows how to get the throughput.
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs:182]:
```suggestion
        /// The following example shows how to replace the throughput.
```

File: Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs [hunk 2]
```diff
@@ -173,14 +173,24 @@ public abstract class Database
         /// The following example shows how to get the throughput.
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs:173]:
```suggestion
        /// The following example shows how to replace the throughput.
```

File: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs [hunk 3]
```diff
@@ -301,10 +301,28 @@ public abstract class Container
...
+        /// The following example shows how to get the throughput.
+        /// <code language="c#">
+        /// <![CDATA[
+        /// ThroughputResponse throughput = await this.cosmosContainer.ReplaceThroughputAsync(
+        ///     ThroughputProperties.CreateFixedThroughput(10000));
+        /// ]]>
+        /// </code>
+        /// </example>
+        /// <example>
+        /// The following example shows how to get the throughput.
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs:314]:
```suggestion
        /// The following example shows how to replace the throughput.
```

File: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs [hunk 4]
```diff
@@ -301,10 +301,28 @@ public abstract class Container
         /// <param name="requestOptions">(Optional) The options for the throughput request.</param>
         /// <param name="cancellationToken">(Optional) <see cref="CancellationToken"/> representing request cancellation.</param>
         /// <returns>The throughput response.</returns>
+        /// <example>
+        /// The following example shows how to get the throughput.
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs:305]:
```suggestion
        /// The following example shows how to replace the throughput.
```

**Comment by jahmai** [comment]
Nice. I will try this out soon in our own preview branch.

File: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs [hunk 5]
```diff
@@ -301,10 +301,28 @@ public abstract class Container
         /// <param name="requestOptions">(Optional) The options for the throughput request.</param>
         /// <param name="cancellationToken">(Optional) <see cref="CancellationToken"/> representing request cancellation.</param>
         /// <returns>The throughput response.</returns>
+        /// <example>
+        /// The following example shows how to replace the fixed throughput.
+        /// <code language="c#">
+        /// <![CDATA[
+        /// ThroughputResponse throughput = await this.cosmosContainer.ReplaceThroughputAsync(
+        ///     ThroughputProperties.CreateFixedThroughput(10000));
```
Comment (deborahc) [inline Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs:309]: We'll be using the word "Manual" in the Portal to differentiate between what we have today and autoscale. It'd be good to align here, e.g. "CreateManualThroughput().

Comment (FabianMeiswinkel) [reply Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs:309]: I would also consistently use or skip Provisioned s- either CreateManuallyProvisionedThroughput and CreateAutomaticallyProvisionedThroughput or CreateManualThroughput/CreateAutoscaleThroughput

**Pull request merged**